presto-spi/src/main/java/com/facebook/presto/spi/block/BlockBuilder.java

```diff
@@ -76,6 +76,15 @@ default BlockBuilder beginBlockEntry()
     }
 
     /**
+     * Create a new block from the current materialized block by keeping the same elements
+     * only with respect to {@code visiblePositions}.
+     */
+    default Block mask(int[] visiblePositions)
```

Comment:
I feel in general it could be useful; functions like `copyRegion` has a similar favor.